### PR TITLE
Fix web concurrency tests

### PR DESCRIPTION
**Pull request by hunterloftis**

### CODE REVIEW

File: profile/nodejs.sh [hunk 1]
```diff
@@ -24,10 +29,13 @@ detect_memory() {
...
 export NODE_HOME="$HOME/.heroku/node"
-export NODE_ENV=${NODE_ENV:-production}
 
 calculate_concurrency
 
 export MEMORY_AVAILABLE=$MEMORY_AVAILABLE
 export WEB_MEMORY=$WEB_MEMORY
 export WEB_CONCURRENCY=$WEB_CONCURRENCY
+
+if [ "$LOG_CONCURRENCY" = "true" ]; then
```
Comment (nmccready) [inline profile/nodejs.sh]: I am getting errors on `$LOG_CONCURRENCY` not existing. Do we need to define this now on all heroku instances? WHy is it not set to false or true by default in environment.sh?

Comment (hunterloftis) [inline profile/nodejs.sh]: Can you explain how to reproduce the issue? nodejs.sh doesn't set `set -o nounset` so unset vars should be fine.

Comment (nmccready) [inline profile/nodejs.sh]: I have another buildpack which expects node to be installed via this webpack. On circlci deployment is failing with the following variable.

`remote: /tmp/build_f9481d5b2938229e585365cbf1fce652/.profile.d/nodejs.sh: line 39: LOG_CONCURRENCY: unbound variable`

Once I set LOG_CONCURRENCY explicitly in heroku settings the error dissapeared. So to me this should be added to the documentation or set via environment.sh if it does not exist.

Comment (nmccready) [inline profile/nodejs.sh]: The dependent buildpack is here https://github.com/realtymaps/heroku-buildpack-assets

Comment (nmccready) [inline profile/nodejs.sh]: Full Error.

```
remote: -----> Creating Asset environment
remote: 
remote: ASSETS_MODULES_CACHE=true
remote: ASSETS_CONFIG_PRODUCTION=true
remote: ASSETS_CACHE_DIRECTORY=_public
remote: ASSETS_COMPILE_CMD=npm run gulp prod
remote: /tmp/build_f9481d5b2938229e585365cbf1fce652/.profile.d/nodejs.sh /tmp/build_f9481d5b2938229e585365cbf1fce652/scripts/app/syncVars
remote: /tmp/build_f9481d5b2938229e585365cbf1fce652/.profile.d/nodejs.sh: line 39: LOG_CONCURRENCY: unbound variable
remote: 
remote:  !     Push rejected, failed to compile Multipack app
```

Comment (hunterloftis) [inline profile/nodejs.sh]: I have not been able to reproduce the issue; none of these logs ("Creating Asset environment, etc") are strings that exist either in this buildpack or in the master branch of the linked buildpack.

It sounds like the other buildpack might be inadvertently trying to source nodejs.sh with `set -o nounset`.

Comment (nmccready) [inline profile/nodejs.sh]: https://github.com/realtymaps/heroku-buildpack-assets/blob/master/bin/compile#L56

Comment (hunterloftis) [inline profile/nodejs.sh]: Ah; forked repos aren't searchable. Is that a new thing from github or have I just never noticed it?

Comment (nmccready) [inline profile/nodejs.sh]: I think that is a bug as I tried searching too and it didnt work.

Comment (hunterloftis) [inline profile/nodejs.sh]: https://github.com/realtymaps/heroku-buildpack-assets/blob/master/bin/compile#L49

Where do `load_dependencies` and `$ASSETS_DEPS` come from?

Comment (nmccready) [inline profile/nodejs.sh]: https://github.com/realtymaps/heroku-buildpack-assets/blob/master/lib/environment.sh#L17-L24

Comment (nmccready) [inline profile/nodejs.sh]: https://github.com/realtymaps/heroku-buildpack-assets/blob/master/lib/environment.sh#L7

Comment (hunterloftis) [inline profile/nodejs.sh]: Ah, here you go: https://github.com/realtymaps/heroku-buildpack-assets/blob/master/lib/environment.sh#L7

That's running `nodejs.sh` from the buildpack. The stuff in .profile.d is not supposed to be run during build - it creates the execution environment. You can get details here:

- https://devcenter.heroku.com/articles/buildpack-api

**Comment by nmccready** [comment]
Ok,  it's still kinda stinks that this worked fine prior to this change.  However adding the various to heroku fixed the issue.

**Comment by hunterloftis** [comment]
What is the assets buildpack designed to do @nmccready?

**Comment by nmccready** [comment]
Builds the client side assets by running gulp or whatever you want.  Basically our project won't rebuild the assets if they exist already.  So the assets are set to build once on deployment. That way restarts, cycles,  etc are fast.

**Comment by hunterloftis** [comment]
Gotcha. You could look into using a `postinstall` script for that:

- https://devcenter.heroku.com/articles/node-with-grunt

You can also cache things like bower deps so they won't have to be re-downloaded/built:

- https://devcenter.heroku.com/articles/nodejs-support#cache-behavior

**Pull request merged**